### PR TITLE
chore: move fjs as dev deps

### PR DESCRIPTION
**Pull request by climba03003**
After updated to `4.2.0`, the generated code do not depends on `fjs`, `ajv`, `ajv-formats`, etc.

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/error-serializer.js [hunk 1]
```diff
@@ -1,15 +1,183 @@
...
+    }
+
+    if (!found) {
+      result = str
+    } else {
+      result += str.slice(last)
+    }
+    return ((point < 32) || (surrogateFound === true)) ? JSON.stringify(str) : '"' + result + '"'
+  }
+}
```
[the remark this answers is not in the document]
Comment (climba03003) [reply lib/error-serializer.js:173]: Yes

**Comment by mcollina** [comment]
Please update the fast-json-stringify-compiler depenendecy too.

File: package.json [hunk 2]
```diff
@@ -181,7 +181,7 @@
     "@fastify/fast-json-stringify-compiler": "^3.0.0",
     "abstract-logging": "^2.0.1",
     "avvio": "^8.1.3",
-    "fast-json-stringify": "^4.1.0",
+    "fast-json-stringify": "^4.2.0",
```
Comment (mcollina) [inline package.json:184]: I understand we can drop this?

File: package.json [hunk 3]
```diff
@@ -181,7 +181,7 @@
     "@fastify/fast-json-stringify-compiler": "^3.0.0",
```
Comment (mcollina) [inline package.json:181]:
```suggestion
    "@fastify/fast-json-stringify-compiler": "^3.0.1",
```

File: package.json (hunk 2 above)
Comment (climba03003) [reply package.json:184]: Yes, it needs to move back as `devDenpendencies`. Forgotten to type `-D` when install it again.

**Review by Eomm — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Review by climba03003 — COMMENTED** [review]
CI is actually passed and I think there are some problem of the newly added workflow
https://github.com/fastify/fastify/blob/main/.github/workflows/md-lint.yml

@Fdawgs do you have time to check what is the problem on it.

**Comment by Fdawgs** [comment]
There is an issue with using the npm cache in the `actions/setup-node` action at the moment: https://github.com/actions/setup-node/issues/516

**Pull request merged**